src/Conveyor/curved_belt_conveyor.gd

```diff
@@ -748,34 +771,45 @@ func _exit_tree() -> void:
 	OIPComms.tag_group_initialized.disconnect(_tag_group_initialized)
 	OIPComms.tag_group_polled.disconnect(_tag_group_polled)
 
-func _notification(what: int) -> void:
-	if what == NOTIFICATION_LOCAL_TRANSFORM_CHANGED:
-		set_notify_local_transform(false)
-		scale = Vector3(scale.x, 1, scale.x)
-		set_notify_local_transform(true)
-		OnScaleChanged()
-
-func OnScaleChanged() -> void:
-	if prev_scale_x != scale.x:
-		RecalculateSpeeds()
-		notify_property_list_changed()
-		if scale.x > 1.0:
-			UpdateBeltMaterialScale()
-			UpdateMetalMaterialScale()
-		var ce1 = get_conveyor_end1()
-		var ce2 = get_conveyor_end2()
-		if ce1:
-			ce1.scale = scale.inverse()
-			ce1.size.z = scale.x
-		if ce2:
-			ce2.scale = Vector3(scale.z, scale.y, scale.x).inverse()
-			ce2.size.z = scale.z
-		prev_scale_x = scale.x
-
+func migrate_scale_to_size():
+	if scale == Vector3.ONE:
+		return 
+	scale = Vector3.ONE
+	size = Vector3(scale.x * SIZE_DEFAULT.x, SIZE_DEFAULT.y, scale.x * SIZE_DEFAULT.x)
+		
+func set_mesh_size(new_size: Vector3) -> void:
+	var adjusted_size: Vector3 = Vector3(new_size.x, SIZE_DEFAULT.y, new_size.x)
+	if size != adjusted_size:
+		size = adjusted_size
+		scale = Vector3(size.x / SIZE_DEFAULT.x, 1, size.x / SIZE_DEFAULT.x)
+		_on_size_changed()
+
+func _on_size_changed() -> void:
+	if not is_inside_tree():
+		return
```

Comment:
FYI, the first call of this happens before we enter the tree, but it should still work if you've duplicated the same logic in `_ready`.